online-shop/src/app/product-list/product-list.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { Product } from '../data/product';
+import { AuthService } from '../services/auth.service';
+import { ProductService } from '../services/product.service';
+
+@Component({
+  selector: 'app-product-list',
+  templateUrl: './product-list.component.html',
+  styleUrls: ['./product-list.component.scss'],
+})
+export class ProductListComponent implements OnInit {
+  products: Product[] = [];
+  canEditProducts: boolean = this.authService.userHasRole('admin');
```

Comment:
**suggestion:** you should have this assignment in the component lifecycle, more specifically in `ngOnInit`